pipeline/article_scraper/AmisScraper/spiders/spiders.py

```diff
@@ -9,56 +9,59 @@
 
 
 class UnicodeFriendlyLinkExtractor(SgmlLinkExtractor):
-    """
+    '''
     Need this to fix the encoding error.
     Taken from https://stackoverflow.com/questions/17862016/scrapy-python-unicode-links-error
-    """
+    '''
+
     def extract_links(self, response):
         base_url = None
         if self.restrict_xpaths:
             hxs = HtmlXPathSelector(response)
             base_url = get_base_url(response)
             body = u''.join(f for x in self.restrict_xpaths
-                           for f in hxs.select(x).extract())
+                            for f in hxs.select(x).extract())
             try:
                 body = body.encode(response.encoding)
             except UnicodeEncodeError:
                 body = body.encode('utf-8')
         else:
             body = response.body
 
-        links = self._extract_links(body, response.url, response.encoding, base_url)
+        links = self._extract_links(
+            body, response.url, response.encoding, base_url)
         links = self._process_links(links)
         return links
 
 
-
 class BloombergSpider(CrawlSpider):
-    name = "bloomberg"
-    allowed_domains = ["bloomberg.com"]
-    start_urls = ["http://www.bloomberg.com"]
+    name = 'bloomberg'
+    allowed_domains = ['bloomberg.com']
+    start_urls = ['http://www.bloomberg.com']
     rules = [
         Rule(UnicodeFriendlyLinkExtractor(allow='(/news/articles/)((?!:).)*$'),
-            callback="parse_item", follow=True),
+             callback='parse_item', follow=True),
         Rule(UnicodeFriendlyLinkExtractor(allow='(/news/)((?!:).)*$'),
-             callback="parse_item", follow=True),
+             callback='parse_item', follow=True),
         Rule(UnicodeFriendlyLinkExtractor(allow='(/articles/)((?!:).)*$'),
-             callback="parse_item", follow=True)
+             callback='parse_item', follow=True)
     ]
 
     def parse_item(self, response):
         item = NewsArticleItem()
         title = response.xpath('//title/text()').extract_first()
         if title == 'Bloomberg':
-            title = response.xpath('//meta[@property="og:title"]/text()').extract_first()
+            title = response.xpath(
+                '//meta[@property="og:title"]/text()').extract_first()
 
         article = response.xpath('//p/text()').extract()
-        self.logger.info("Scraping Title: " + title)
+        self.logger.info('Scraping Title: ' + title)
         try:
             item['title'] = title
             item['article'] = article
-            item['link'] = response.url.replace('http://', '').replace('https://', '')
-            raw_date = response.url.split("/")[-2]
+            item['link'] = response.url.replace(
+                'http://', '').replace('https://', '')
+            raw_date = response.url.split('/')[-2]
```

Comment:
Gosh man!! You really formatted even the quotes properly everywhere!!!!

Reply:
My editor has very little tolerance for inconsistencies.... :P

Reply:
+1 for that.